src/html.rs

```diff
@@ -400,7 +400,7 @@ where
                         plain
                     } else {
                         stack.push((node, false, Phase::Post));
-                        self.format_node(node, true)?
+                        !self.format_node(node, true)?
```

Comment:
This is an interesting change; can you elaborate on it?

Reply:
(Also, to be clear, if it's a mistaken change, then the real mistake is that no test failure resulted from it! And if it isn't, we should add a regression test! Either way it's interesting.)

Reply:
Ah sorry, I meant to comment on that change.

All the new `render_` functions return a `io::Result<bool>`, and return an `Ok(true)` when the function is successful. The only render function that can return an `Ok(false)` is `render_image`. And that `false` needs to trigger some additional work, just like it did before.

Originally `format_node` was returning `Ok(false)` by default, which would set `new_plain` in `format`.

But with the render functions returning a `true` for success, then it made more sense to set a `processing_complete` boolean with value of the render functions, and return that.  So now we had to negate the return value in the line above.

Reply:
The tests do fail pretty spectacularly if the `!` is removed.

Reply:
Ah, got it; I misread the diff and thought the `true`s and `false`s hadn't in fact switched positions. I might change those to a binary enum that actually make clear what work they're doing.